### ISSUE TITLE
Reading same line multiple times

### ISSUE THREAD

**Issue by sami-sweng** [nickewing/line-reader#15]
When multiple workers use the same reader in parallel, with hasNextLine() and nextLine(), some lines are read multiple times.

**Comment by lfdoherty** [comment]
Just making a note here that this was the true cause of the problem I logged as bug #21. If it's not going to be fixed I strongly suggest a note be added to the documentation warning that it's unsafe to do this.

**Comment by lfdoherty** [comment]
I should also add that it's not as simple as some lines being read multiple times - text may be mangled within a line (possibly if it spans more than one data chunk?)

**Comment by amitguptagwl** [comment]
I found a bug today on the same line. If I created a file with single line, lineReader.eachLine() keeps running forever.

```
var lineReader = require('line-reader');

lineReader.eachLine('testfile', function(line, last) {
	console.log(line);
});
```
But if there are more than one line even if they are empty. it works fine.

**Comment by nickewing** [comment]
@amitguptagwl What version of line-reader are you using?  There are tests for both a file with a single line and a new line character and a file with a single line and no new line character.  Are you also using workers like the previous posters?

**Comment by amitguptagwl** [comment]
I am using "line-reader": "^0.3.0". I dint realized that there is a new version available. Let me update and try again.

**Comment by amitguptagwl** [comment]
Thanks It is fixed in 0.4.0

**Comment by nickewing** [comment]
Coo,l glad to hear it. :+1:
